Mage.Verify/src/test/java/mage/verify/VerifyCardDataTest.java

```diff
@@ -1852,6 +1856,34 @@ private void compareClassRecursive(Object obj1, Object obj2, Card originalCard,
         }
     }
 
+    // One (fake) event per batch event type
+    private static final Set<GameEvent> fakeBatchEvents;
+
+    static {
+        fakeBatchEvents = Stream
+                .of(GameEvent.EventType.values())
+                .filter(GameEvent.EventType::isBatch)
+                .map(eventType -> new GameEvent(eventType, null, null, null))
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Perform checks on abilities
+     */
+    private void checkAbility(Card originalCard, Ability ability, String msg) {
```

Comment:
Must remove such code after all fixes. It’s too hacky.

Reply:
Done